src/common/pf_ppm.c

```diff
@@ -188,42 +194,91 @@ static void pf_ppm_finish_buffer(
    memcpy(&p_payload[p_ppm->transfer_status_offset], &p_ppm->transfer_status, sizeof(p_ppm->transfer_status));
 }
 
+/**
+ * @internal
+ * Calculate the delay to use with the scheduler, by taking the stack cycle time into account.
+ *
+ * With a stack cycle time of 1 ms, a scheduled delay of 0-700 microseconds will
+ * cause a nice periodicity of 1 ms. A scheduled delay of 1000 microseconds will
+ * sometimes fire at next cycle, sometimes not. This gives an event spacing of
+ * 1 or 2 ms.
+ *
+ * Similarly a scheduled delay of 1100 to 1700 microseconds causes
+ * a nice periodicity of 2 ms, and a scheduled delay of 2100 to 2700
+ * microseconds causes a nice periodicity of 3 ms. These measurements were
+ * made using a Ubuntu Laptop.
+ *
+ * Scheduling a delay close to a multiple of the stack cycle time is risky, and
+ * should be avoided. Calculate the number of stack cycles to wait, and
+ * calculate a delay corresponding to half a cycle less.
+ *
+ * If the requested delay is in the range 1.5 to 2.5 stack cycle times, this
+ * function will return a calculated delay giving a periodicity of 2 stack cycle
+ * times. If the requested time is less than 1.5 stack cycle times, the
+ * resulting periodicity is 1 stack cycle time.
+ *
+ * Note that this function calculates the delay time required to make the
+ * scheduler fire at a specific stack tick. However the time jitter in the
+ * firing is largely dependent on the underlaying operating system's ablitity to
+ * trigger the stack execution with a high time precision.
+ *
+ * @param wanted_delay     In:    Wanted delay in microseconds.
+ * @param stack_cycle_time In:    Stack cycle time in microseconds. Must be larger than 0.
+ * @return Number of microseconds of delay to use with the scheduler.
+ */
+uint32_t pf_ppm_calculate_compensated_delay(
+   uint32_t                wanted_delay,
+   uint32_t                stack_cycle_time
+)
+{
+   uint32_t                number_of_stack_tics = 1;  /* We must wait at least one tick */
```

Comment:
Change "number_of_stack_tics" to "number_of_stack_ticks"

Reply:
Done